src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -382,8 +397,11 @@ private async Task<Compilation> GetOrBuildDeclarationCompilationAsync(SolutionSt
                             return compilation;
                         }
 
+                        TrackedGeneratorDriver generatorDriver;
+                        (compilation, generatorDriver) = await BuildDeclarationCompilationFromInProgressAsync(solutionServices, (InProgressState)state, compilation, cancellationToken).ConfigureAwait(false);
+
                         // We must have an in progress compilation. Build off of that.
-                        return await BuildDeclarationCompilationFromInProgressAsync(solution, (InProgressState)state, compilation, cancellationToken).ConfigureAwait(false);
+                        return compilation;
```

Comment:
odd that the driver is dropped on the floor.. can we doc?

Reply:
Ah right it seems this was only used for skeleton assembly reference generation which is a bit funky. I'm going to file a bug to delete this entire function since I think there's some cleanup we can do here.

Reply:
https://github.com/dotnet/roslyn/issues/42819 filed. This code all looks fishy in general... :-(